plugins/inputs/t128_tank/reader.go

```diff
@@ -270,6 +277,10 @@ func (r *Reader) readFromTank(readCtx context.Context, startingIndex index) (err
 		for _, message := range messages {
 			collectedMessages = append(collectedMessages, *message)
 		}
+		if len(collectedMessages) > 0 {
```

Comment:
Please reference the intel agent's `128intelagent/events/read.go`. Saving on every message will completely destroy our ability to process messages quickly. Replicate what's done there with a ticker, save on shutdown, and save on index multiples of 1k.